exchange/exchange.go

```diff
@@ -85,6 +85,21 @@ func (e *exchange) HoldAuction(ctx context.Context, bidRequest *openrtb.BidReque
 		}
 	}
 
+	for _, impInRequest := range bidRequest.Imp {
```

Comment:
Each time through the loop, we will be partially overwriting the data from the previous Imp with the next Imp.

I see two possible ways forward, from inside the loop call a per-Imp metrics method to record these metrics. Or turn these booleans into `int`s and increment them each iteration of the loop. Then have the base request metrics engine method advance the counters by the values stored as ints.

Reply:
Hans, I was precisely working in this PR right now nad just pushed an update where this loop changes. (The entire solution does for that matter). Can you please refresh? Prometheous implementation is still not ready though but Influx probably is unless I'm missing something